src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1186,6 +1234,16 @@ LLVMValueRef ShadowStackTop
 
         private void EmitNativeToManagedThunk(WebAssemblyCodegenCompilation compilation, MethodDesc method, string nativeName, LLVMValueRef managedFunction)
         {
+            if (_pinvokeMap.TryGetValue(nativeName, out MethodDesc existing))
```

Comment:
This method gets used for RuntimeExports (and eventually for NativeCallable methods). Why are you adding them to the P/Invoke map?

Reply:
They have to be added because they can collide. Because they are unmangled C extern type names, If I have a RuntimeExport that maps to fred and a (real) pInvoke that maps to fred, I will run into problems if fred doesn't have exactly the same signature in both.